test/plugins/node-workspace.ts

```diff
@@ -0,0 +1,2014 @@
+      const nodeWS = new NodeWorkspaceDependencyUpdates(
+        github,
+        config,
+        'node-workspace',
+        checkpoint
+      );
+      const [actualManifest, actualChanges] = await nodeWS.run(
+        newManifestVersions,
+        pkgsWithPRData
+      );
+      mock.verify();
+      expect(logs).to.eql([
+        [
+          'node-workspace: found packages/pkgA/package.json in changes',
+          CheckpointType.Success,
+        ],
+        [
+          'node-workspace: loaded packages/pkgA/package.json from existing changes',
+          CheckpointType.Success,
+        ],
+        [
+          'node-workspace: loaded packages/pkgB/package.json from github',
+          CheckpointType.Success,
+        ],
+        [
+          'node-workspace: loaded packages/pkgC/package.json from github',
+          CheckpointType.Success,
+        ],
+        [
+          'node-workspace: setting packages/pkgA/package.json to 1.1.2 from release-please',
+          CheckpointType.Success,
+        ],
+        [
+          'node-workspace: setting packages/pkgB/package.json to 2.2.3 from dependency bump',
+          CheckpointType.Success,
+        ],
+        [
+          'node-workspace: setting packages/pkgC/package.json to 3.3.4 from dependency bump',
+          CheckpointType.Success,
+        ],
+        [
+          'node-workspace: @here/pkgB.@here/pkgA updated to ^1.1.2',
+          CheckpointType.Success,
+        ],
+        [
+          'node-workspace: @here/pkgC.@here/pkgB updated to ^2.2.3',
+          CheckpointType.Success,
+        ],
+      ]);
+
+      expect([...actualManifest]).to.eql([
+        ['packages/pkgA', '1.1.2'],
+        ['py/pkg', '1.1.2'],
+        ['packages/pkgB', '2.2.3'],
+        ['packages/pkgC', '3.3.4'],
+      ]);
+      const expected = [
+        pkgAData,
+        pyPkgData,
+        {
+          config: {
+            releaseType: 'node',
+            packageName: '@here/pkgB',
+            path: 'packages/pkgB',
+          },
+          prData: {
+            version: '2.2.3',
+            changes: new Map([
+              [
+                'packages/pkgB/package.json',
+                {
+                  content: packageJsonStringify({
+                    name: '@here/pkgB',
+                    version: '2.2.3',
+                    dependencies: {
+                      '@here/pkgA': '^1.1.2',
+                      someExternal: '^9.2.3',
+                    },
+                  }),
+                  mode: '100644',
+                },
+              ],
+              [
+                'packages/pkgB/CHANGELOG.md',
+                {
+                  content:
+                    '# Changelog' +
+                    '\n\nAll notable changes to this project will be ' +
+                    'documented in this file.' +
+                    '\n\n### [2.2.3](https://www.github.com/fake/repo/compare' +
+                    '/pkgB-v2.2.2...pkgB-v2.2.3) (1983-10-10)' +
```

Comment:
I think that these assertions are going to become a bit fragile, as even subtle changes to the markdown generation will lead to a bunch of fiddling in tests.

Having confirmed the behavior manually (i.e., now that you trust the approach), I would be tempted to move the assertion into a snapshot -- at which point, we should just see small stylistic differences in the snapshot as changes are made to content.

Reply:
absolutely. converted the log tracking and the changes to snapshots